standard/expressions.md

```diff
@@ -4562,7 +4565,19 @@ The conditional operator is right-associative, meaning that operations are group
 
 The first operand of the `?:` operator shall be an expression that can be implicitly converted to `bool`, or an expression of a type that implements `operator true`. If neither of these requirements is satisfied, a compile-time error occurs.
 
-The second and third operands, `x` and `y`, of the `?:` operator control the type of the conditional expression. If both `x` and `y` are *default_literal*s ([§12.8.20](expressions.md#12820-default-value-expressions)), a compile-time error occurs.
+If `ref` is present:
+Both *expression*s shall be variables having the same type, and that type is the type of the result.
```

Comment:
How about different types but with an identity conversion?

```csharp
dynamic d = "";
object o = "";
bool condition = true;
ref object x = ref (condition ? ref o : ref d);
```

Reply:
Updated.

Added text for implicitly typed variables (which prefers dynamic).